python_modules/dagster/dagster/_check/__init__.py

```diff
@@ -1233,6 +1233,20 @@ def opt_nullable_iterable_param(
     return iterable_param(obj, param_name, of_type, additional_message)
 
 
+def is_iterable(
+    obj: object,
+    of_type: Optional[TTypeOrTupleOfTTypes[T]] = None,
+    additional_message: Optional[str] = None,
+) -> Iterable[T]:
+    if not isinstance(obj, Iterable):
```

Comment:
these isinstance checks against collections abstract base classes can be slow, but only matters if this ends up on a hot path

Reply:
should we short-circuit if the incoming thing is a list or tuple?

```python
    if isinstance(obj, (list, tuple)):
        return obj
```

Reply:
yep, i did `type(x) is XXX` in some other places here in `check`

```
$ python -m timeit -s "import typing" -- "isinstance([], typing.Iterable)"
1000000 loops, best of 5: 270 nsec per loop
$ python -m timeit -s "import typing" -- "isinstance([], (tuple, list))"
5000000 loops, best of 5: 50.4 nsec per loop
$ python -m timeit -s "import typing" -- "type([]) in (list, tuple)"
10000000 loops, best of 5: 35.6 nsec per loop
$ python -m timeit -s "import typing" -- "type([]) is list"
20000000 loops, best of 5: 19.9 nsec per loop
```